metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/execution/WorkflowExecutor.java

```diff
@@ -232,35 +247,51 @@ private void periodicCheckingLoop(long sleepTime, AbstractMetisPlugin abstractMe
             monitorResult.getTaskState() == TaskState.REMOVING_FROM_SOLR_AND_MONGO
                 ? PluginStatus.CLEANING : PluginStatus.RUNNING);
         workflowExecution.setUpdatedDate(updatedDate);
-        workflowExecutionDao.updateMonitorInformation(workflowExecution);
       } catch (InterruptedException e) {
         LOGGER.warn("Thread was interrupted during monitoring of external task", e);
         Thread.currentThread().interrupt();
         return;
       } catch (ExternalTaskException e) {
-        consecutiveCancelOrMonitorFailures++;
-        LOGGER.warn(String.format("Monitoring of external task failed %s/%s",
-            consecutiveCancelOrMonitorFailures, MAX_CANCEL_OR_MONITOR_FAILURES), e);
-        if (consecutiveCancelOrMonitorFailures == MAX_CANCEL_OR_MONITOR_FAILURES) {
-          monitorResult = new MonitorResult(TaskState.DROPPED, e.getMessage());
-          break;
+        if (ExternalRequestUtil
+            .doesExceptionCauseMatchAnyOfProvidedExceptions(mapWithRetriableExceptions, e)) {
+          consecutiveCancelOrMonitorFailures++;
+          LOGGER.warn(String.format(
+              "Monitoring of external task failed %s. After exceeding %s retries, pending status will be set",
```

Comment:
Comment may not be clear to readers... perhaps add the text 'consecutive times' after the first %s?

Reply:
Fixed